### ISSUE TITLE
Proposal: use iced-x86 for your x86 decoding

### ISSUE THREAD

**Issue by woodruffw** [pkgw/elfx86exts#47]
Hey there! A coworker of mine shared this with me, and I thought it was very cool. 

I noticed that you're using Capstone for your instruction decoding, and then mapping Capstone's internal "group" enum back to human-readable identifiers. I wanted to propose that you give [iced-x86](https://github.com/icedland/iced) a spin (no affiliation!); I think it has several features that you'll find advantageous:

1. It has direct support for CPUID/feature-level extraction, via [`cpuid_features`](https://docs.rs/iced-x86/1.11.0/iced_x86/struct.Instruction.html#method.cpuid_features)
2. It's written in pure Rust, which means that your users won't need to install `libcapstone`
2. It's much more correct than Capstone (from [fuzzing](https://blog.trailofbits.com/2019/10/31/destroying-x86_64-instruction-decoders-with-differential-fuzzing/)), about on par with Intel's reference decoder (XED)

I'm happy to help with the work that would be involved in replacing Capstone, if you're interested! Feel free to close otherwise 🙂

**Comment by pkgw** [comment]
Hi — thanks for your interest!

The short story is that I am super busy these days and this tool is a very low priority for me, so this I am definitely not going to undertake this work myself anytime soon. But if someone's interested and wants to attempt this switch, I am more than happy to accept contributions! I'll leave this issue open for tracking purposes.

**Comment by HanabishiRecca** [comment]
I don't think it is worth doing here as it effectively requires rewriting the app from scratch. Latest fixes made the app perform well enough. Also ARM support is a thing.

I started a standalone X86-only tool instead: https://github.com/HanabishiRecca/bin-cpuflags-x86
Using `iced-x86` it is 20-30x faster according to my tests.

**Comment by pkgw** [comment]
Impressive! @HanabishiRecca I am more than happy to update the README and other materials to recommend people to use your tool if/when it meets their needs — seeing as I only aim to put minimal maintenance effort into this tool, I absolutely to not mind any "competition". Feel free to submit a pull request on the topic if you'd like.

**Comment by HanabishiRecca** [comment]
Thank you. I think this project can benefit more from wider architectures support. As far as I can tell, at this point it downs to simply mapping existing Capstone groups to human-readable aliases.
